src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Handles read/write calls to hard disk.
+ */
+public class Storage {
+    private Path saveFile;
+    private Ui ui;
+
+    public Storage(Path saveFile, Ui ui) {
+        this.saveFile = saveFile;
+        this.ui = ui;
+    }
+
+    /**
+     * Initialises tasks from hard drive.
+     *
+     * @param tasks The place tasks are stored.
+     */
+    public void initialiseTasks(ArrayList<Task> tasks) {
+        try {
+            File sf = new File(String.valueOf(saveFile));
+            if (Files.exists(saveFile)) {
+                Scanner taskReader = new Scanner(sf);
+                while (taskReader.hasNextLine()) {
+                    String taskln = taskReader.nextLine();
+                    String[] taskComps = taskln.split("\\|");
+                    if (taskComps[0].equals("T")) {
+                        tasks.add(new Todo(taskComps[2], Boolean.parseBoolean(taskComps[1])));
+                    } else if (taskComps[0]. equals("D")) {
```

Comment:
No need a space here
```suggestion
                    } else if (taskComps[0].equals("D")) {
```